packages/router/__tests__/router-test.ts

```diff
@@ -1919,6 +1945,30 @@ describe("a router", () => {
       });
     });
 
+    it("reloads all routes after action redirect (chained redirects)", async () => {
+      let t = initializeTmTest();
+      let A = await t.navigate("/foo", {
+        formMethod: "post",
+        formData: createFormData({ gosh: "dang" }),
+      });
+      expect(A.loaders.root.stub.mock.calls.length).toBe(0);
+
+      let B = await A.actions.foo.redirectReturn("/bar");
+      expect(B.loaders.root.stub.mock.calls.length).toBe(1);
+
+      await B.loaders.root.resolve("ROOT*");
+      let C = await B.loaders.bar.redirectReturn("/baz");
+      expect(C.loaders.root.stub.mock.calls.length).toBe(1);
```

Comment:
This second redirect should carry forward the fact that the previous was an action reload and run all loaders